### DISCUSSION TITLE
Dynamic Trigger

### DISCUSSION THREAD

**Discussion by michaelmairegger** [quartznet/quartznet#1279] [Q&A] [answered]
I want to achieve a file change listener trigger that dynamically triggers on file creation/modification under a predefined directory.

I tried a bit but as far as I understand the next job trigger must be **predefined** and cannot be **unknown**, is this correct? How do I create dynamic job executions?

**Comment by lahma** [comment, marked as answer]
Have you checked how [XML-based scheduling](https://github.com/quartznet/quartznet/blob/main/src/Quartz.Plugins/Plugin/Xml/XMLSchedulingDataProcessorPlugin.cs) works? It has logic to poll for file changes and re-schedule as necessary.

**Comment by michaelmairegger** [comment]
Thanks, that helped me for finding a solution
